### ISSUE TITLE
Loading AMD modules fails when initiating a load in `setTimeout`

### ISSUE THREAD

**Issue by lddubeau** [systemjs/systemjs#1386]
### Versions

At the time of submitting the issue:

systemjs: 0.19.35
systemjs-builder: 0.15.25

### Steps to reproduce the problem

1. Clone this repo: https://github.com/lddubeau/systemjs_issue1

2. ``npm install``.

3. ``npm run build`` will produce ``js/out.js``.

4. Open ``index.html`` in a server. (You cannot just load it from ``file://`` due to cross-origin restrictions.)

You may need to reload, or play with the time parameter of `setTimeout` to get an error, the problem is a race condition.

The `index.html` contains the following:

```
<html>
  <head>
    <script type="text/javascript" src="node_modules/systemjs/dist/system.src.js"></script>
    <script type="text/javascript" src="system-config.js"></script>
    <script type="text/javascript">
      System.config({
      bundles: {
        "out.js": [ "lib/modB.js"],
      }
      });
      window.require = SystemJS.amdRequire;
      window.define = SystemJS.amdDefine;
      require(["modA"]);
      setTimeout(function () {
        require(["modC", "lib/modB"]);
      }, 10);
    </script>
  </head>
  <body>
  </body>
</html>
```

The `system-config.js` file contains:

```
System.config({
    baseURL: "js",
    packages: {
        '': {
            defaultExtension: "js"
        }
    }
});
```

### Expected behavior

Everything should load without error. Files should be loaded once and only once.

### Actual behavior

`modC.js` fails to load properly. Two HTTP requests are made for it, and its loading ends with an error.

Errors on the console:

```
Assertion failed: is loading Uncaught (in promise)
Error: (SystemJS) Unable to load dependency http://localhost:3000/js/modC.js.(…)
Uncaught (in promise) Error: (SystemJS) load.execute is not a function(…)
```

I've also seen an error triggered by `normalizedDeps` being `null` at [this line](https://github.com/systemjs/systemjs/blob/3d097d90485a3ceaf5751b5b2709ab7606502105/dist/system.src.js#L3086).

### Observations

I ran into this problem while trying to port a large application based on RequireJS to SystemJS. The code base has worked fine with RequireJS for years. It also works with SystemJS so long as I don't try to load a bundle that calls out to a module that lives outside the bundle.

**Comment by guybedford** [comment]
Thanks so much for the very clear replication.

It seems like the assertion itself is invalid and removing the assertion fixes the issue. I've included this in https://github.com/systemjs/systemjs/commit/a43201f8b2bb354dbc4776630b8ba4359edc7098.

The assertion comes from the original loader spec, which did break a few invariants as it wasn't well-tested for these kinds of conditions.

If you run against `system.js` which is minified without the assertions, the case passes as well.

**Comment by lddubeau** [comment]
Thank you for the quick response.

I'll test this as soon as I get a window of opportunity for doing so. I'm going to run it with my full-fledged application and see how it behaves.

Let me ask a question in the meantime. In my initial report, I mentioned also getting an error due to `normalizedDeps` being `null` at [this line](https://github.com/systemjs/systemjs/blob/3d097d90485a3ceaf5751b5b2709ab7606502105/dist/system.src.js#L3086). I *have* seen this error with the same code I've put here in my report but it has been *extremely* hard to get *this specific `normalizedDeps` error* to reproduce. So in the off chance that the errors were related and had the same fix, I focused on the error I was able to reproduce easily. I had figured (incorrectly, it seems) that the internal state of SystemJS was off and once the state was put back in order, all errors would go away.

Now, however, I'm wondering whether there are 2 issues triggered by similar scenarios, because removing the assert does not appear to me that it would fix the `normalizedDeps` issue. Is there something I'm missing?
